pkg/controller/service/servicecontroller.go

```diff
@@ -488,6 +491,9 @@ func (s *ServiceController) needsUpdate(oldService *api.Service, newService *api
 			return true
 		}
 	}
+	if !reflect.DeepEqual(oldService.Annotations, newService.Annotations) {
```

Comment:
I don't think we'll have an order problem here, can you confirm? (i.e we get [src-range1, src-range2] on first update, and [src-rance2, src-range1] on second update).